### PR TITLE
feat(deps): remove jquery from dependencies

### PR DESCRIPTION
**Pull request by lasjorg**
#### Pre-Submission Checklist

- [x] Your pull request targets the `master` branch of freeCodeCamp/testable-projects-fcc
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] Your changes have been tested either locally or using a newly created CDN based on your fork's testable-projects-fcc/build/bundle.js file

#### Type of Change

- [ ] Small bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.

- [ ] Closes currently open issue: Closes #XXX

#### Description

All tests are passing and nothing seems broken but I need some more eyes on this.

I'm a bit iffy on:

* Async fetch in tests is done correctly/tested correctly.

* Feature parity with the jQuery hidden selector. Not sure it matters, the current code seems to work as the old code did. Not sure the width/height check ever really worked as you can set the width/height to 0 and have overflow hidden and the old tooltip tests still pass (as does the new). The boolean logic is also a bit confusing (no surprise there).

The rest I think should be pretty one-to-one when comparing the jQuery code to the plain JS implementation.

---

Notes:

1. Bundle size down from 1,00 MB (1.057.784 bytes) to 947 KB (969.750 bytes). This isn't all that amazing but it's still less than before we added markedjs as a dependency.

2. When working on this I noticed that some of the keyboard shortcuts for the test are using browser shortcuts and should be changed. Not that anyone actually knows about the test keyboard shortcuts I'm guessing.

* `Shift + Ctrl + T` is reopen closed tab in browsers (Firefox/Chrome). Used by test to Open/close test modal.

* `Shift + Ctrl + O` is open/managed bookmarks (Firefox/Chrome). Used by test to Open/close foldout menu.

### CODE REVIEW

File: src/project-tests/tribute-page-tests.js [hunk 1]
```diff
@@ -47,9 +46,8 @@ export default function createTributePageTests() {
       it(`Within the "img-div" element, I should see an <img> element
       with a corresponding id="image".`, function () {
         assert.isNotNull(document.getElementById('image'));
-        assert.strictEqual(
-          $('#img-div').find('#image').length,
-          1,
+        assert.isNotNull(
+          document.querySelector('#img-div #image'),
```
Comment (ShaunSHamilton) [inline src/project-tests/tribute-page-tests.js:50]: Should we add a test that there is not more than 1 `#image` element?

**Review by ShaunSHamilton — COMMENTED** [review]
Thank you, for this, Lasse.

This LGTM, other than minor comment below

**Comment by lasjorg** [comment]
Both `querySelector` and `.find()` will just return the first element with the `id`. It's only if you pass `.find()` something other than an id for the selector (like a type or class) that it will return multiple elements.

Pretty sure the length check in the old code is just because jQuery doesn't return `null` if it doesn't find the element, unlike `querySelector`, and the jQuery object that is returned has a length property you can look at. So what it is really doing is just checking that the length isn't `0`. But written in a confusing way.

If you read the assert messages and requirements you can also see that the requirements are just for if the two elements are descendants of the parent. Failing the two tests because of multiple descendant elements with the same `id` would just be confusing as that is not what the assert message nor requirements are verbalizing.

**Comment by lasjorg** [comment]
I don't see why as that isn't the requirement.

**Comment by ShaunSHamilton** [comment]
> I don't see why as that isn't the requirement.

I am saying to add the requirement. Currently, Campers cannot pass the tests, if they have those `id`s duplicated. However, with this change, Campers can pass with duplicate `id`s.

So, to keep consistency with what past/current/previous projects pass all tests, I am suggesting adding tests for such.

**Comment by lasjorg** [comment]
Not sure if I think that is really necessary.

But if we do want to add a new test I don't think this PR is the correct place to do it.

**Comment by ShaunSHamilton** [comment]
@scissorsneedfoodtoo What is your recommendation here?

**Comment by nhcarrigan** [comment]
My two cents:

I agree with Shaun that we should probably look in to testing for duplicate IDs, especially since (I *believe*) we teach that IDs should be unique.

However, I'm with Lasse that this change is out of scope for this PR, given that the current jQuery tests do not assert this. This PR removes jQuery, but does not change the functionality of the tests, and IMHO that's as it should be.

**Review by naomi-lgbt — APPROVED** [review]

**Review by scissorsneedfoodtoo — APPROVED** [review]
Hey @lasjorg, wow, thank you for taking the time to remove jQuery from all of these tests and 
 updating the main README file. It'll be great to move away from a library that's no longer necessary, and decreasing the bundle size is always nice.

Also, thank you for your helpful insights above. I didn't know that `.find()` worked like that.

I agree with @nhcarrigan's  sentiments -- it would be good to add a test for duplicate ids. I believe we do state in the new curriculum that ids should be unique, and testing the projects for duplicate ids will help reinforce that. But I also agree that doing so is outside the scope of this PR.

It's no problem merging this as-is, and then adding tests in another PR.

**Comment by lasjorg** [comment]
@ShaunSHamilton @nhcarrigan @scissorsneedfoodtoo

Then we have to add the id test, and update the challenge page text, for all the challenges. Would it be only for the RWD pages or all the Front End Development Libraries Projects as well?

I would suggest having a single requirement of no duplicate ids and not tying it to any specific user stories.

> My tribute page should not contain duplicate `id` values

```
it('My tribute page should not contain duplicate `id` values', function () {
  assert.isTrue(
    hasUniqueIdCount(),
    'id attribute values should be unique'
  );
});
```

Do we need to open an issue for this? I do have some code ready-ish I can push.
